pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -343,20 +401,67 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 
 	c.DeviceID, c.DeviceIDTags = buildDeviceID(c.getDeviceIDTags())
 
-	subnet, err := getSubnetFromTags(c.InstanceTags)
-	if err != nil {
-		log.Debugf("subnet not found: %s", err)
-	}
-	c.Subnet = subnet
+	c.ResolvedSubnetName = c.getResolvedSubnetName()
 
 	c.addUptimeMetric()
 	return c, nil
 }
 
+func (c *CheckConfig) getResolvedSubnetName() string {
+	var resolvedSubnet string
+	if c.Network != "" {
+		resolvedSubnet = c.Network
+	} else {
+		subnet, err := getSubnetFromTags(c.InstanceTags)
+		if err != nil {
+			log.Debugf("subnet not found: %s", err)
+		} else {
+			resolvedSubnet = subnet
+		}
+	}
+	return resolvedSubnet
+}
+
+// DiscoveryDigest returns a hash value representing the minimal configs used to connect to the device
+func (c *CheckConfig) DiscoveryDigest(address string) string {
+	h := fnv.New64()
+	// Hash write never returns an error
+	h.Write([]byte(address))                   //nolint:errcheck
+	h.Write([]byte(fmt.Sprintf("%d", c.Port))) //nolint:errcheck
+	h.Write([]byte(c.SnmpVersion))             //nolint:errcheck
+	h.Write([]byte(c.CommunityString))         //nolint:errcheck
+	h.Write([]byte(c.User))                    //nolint:errcheck
+	h.Write([]byte(c.AuthKey))                 //nolint:errcheck
+	h.Write([]byte(c.AuthProtocol))            //nolint:errcheck
+	h.Write([]byte(c.PrivKey))                 //nolint:errcheck
+	h.Write([]byte(c.PrivProtocol))            //nolint:errcheck
+	h.Write([]byte(c.ContextName))             //nolint:errcheck
+
+	// Sort the addresses to get a stable digest
+	addresses := make([]string, 0, len(c.IgnoredIPAddresses))
```

Comment:
should this be `ignoredAddresses` or `ignoredIPs`

Reply:
I would prefer keeping the current naming to match `ignored_ip_addresses` config.

Reply:
Sure renamed here 8040c0e6a